content/blog/serverless-api/index.md

```diff
@@ -0,0 +1,373 @@
+    source_arn=api_gateway.execution_arn.apply(lambda arn: f"{arn}/*/*")
+)
+
+# Export the API Gateway URL
+pulumi.export("url", api_gateway.api_endpoint)
+```
+
+Let's break down what's happening here:
+
+1. **API Gateway**: We create an HTTP API (v2) that will receive all web requests and forward them to our Lambda.
+2. **IAM Role**: Lambda needs permissions to execute and write logs. We create a role with the minimum necessary permissions.
+3. **Container Registry**: We create an ECR repository and use Pulumi's `awsx` package to automatically build and push our Docker image.
+4. **Lambda Function**: We create the Lambda function using our container image. Notice how we set memory and timeout - these are important for a web application.
+5. **Integration**: We connect API Gateway to Lambda using the AWS\_PROXY integration type. `ANY /{proxy+}` means we route everything through with the full path.
+
+Want to add a custom domain? Here's how:
+
+```python
+# Create a certificate
+certificate = aws.acm.Certificate("api-cert",
+    domain_name="api.yourdomain.com",
+    validation_method="DNS"
+)
+
+# Create domain name in API Gateway
+domain_name = aws.apigatewayv2.DomainName("api-domain",
+    domain_name="api.yourdomain.com",
+    domain_name_configuration=aws.apigatewayv2.DomainNameDomainNameConfigurationArgs(
+        certificate_arn=certificate.arn,
+        endpoint_type="REGIONAL",
+        security_policy="TLS_1_2"
+    )
+)
+
+# Map the domain to our API stage
+domain_mapping = aws.apigatewayv2.ApiMapping("api-mapping",
+    api_id=api_gateway.id,
+    domain_name=domain_name.id,
+    stage=stage.id
+)
+```
+
+Setup some Route53 records and cname a domain name appropratelly and you'll have a container backed by a custom domain name.
+
+## Why I love this approach
+
+The really cool thing about using Pulumi and lambdas is that it makes complex infrastructure changes safe and repeatable.
+
+- Want to increase the Lambda memory? Change one number and run `pulumi up`.
+- You get [complete AWS Lambda and API Gateway management](/registry/packages/aws/) for your low-traffic REST APIs
+- And [Infrastructure testing capabilities](/docs/iac/concepts/testing/) mean you can validate your serverless configuration before deployment
+
+You can find the full code in the [service status monitor repo](https://github.com/adamgordonbell/service-status-monitor) from my [python devops article](https://www.pulumi.com/blog/python-for-devops/). A larger version with rolling updates is coming soon.
+
+{{< notes type="info" >}}
+
+## **Alternative Approaches**
+
+There are other ways to run containers that scale to zero. [Google Cloud Run](https://www.pulumi.com/registry/packages/gcp/api-docs/cloudrun/service/) offers similar functionality on GCP, and [AWS App Runner](https://www.pulumi.com/blog/deploy-applications-with-aws-app-runner/) is another AWS service that can do this. Both have similar pricing models—very cheap for low-volume services.
```

Comment:
There is an alternative framework to Pulumi which is using SST which is under the hood using Pulumi

Reply:
yeah, true, SST is a alternative way to setup hosting a backend on lambda. I feel like it's existence shows that people need easy way to do this. But SST is typescript isn't it and my solution here shows doing it in python and hosting anything you can put in a container.  I mean, its not as simple as SST but hopefully valuable to python people.